sdk/core/Azure.Core/src/Shared/MutableJsonDocument.ChangeTracker.cs

```diff
@@ -135,6 +137,51 @@ internal IEnumerable<MutableJsonChange> GetRemovedProperties(string path, int hi
                 }
             }
 
+            internal IEnumerable<string> GetChangedProperties()
```

Comment:
could this routine simply trim the original _changes list? the duplicate entries are garbage, aren't they?

Reply:
I'm still not ready to start pruning the change list, but we can revisit that as we continue to improve perf here - I'm not opposed fundamentally, but I think when we do that, we'll need to update the high water mark logic in other places.